applications/sam/sam_ls.py

```diff
@@ -211,22 +211,24 @@ def _printItems(self, items, args, detectionMethod, filters, level=0):
 
         for item in sorted(items):
             itemType = item.getType()
-            toPrint = True
+            toPrint = False
 
-            # sam-ls -d
-            if args.directories and itemType != sequenceParser.eTypeFolder:
-                toPrint = False
-
-            # sam-ls -f
-            if args.files and itemType != sequenceParser.eTypeFile:
-                toPrint = False
-
-            # sam-ls -s
-            if args.sequences and itemType != sequenceParser.eTypeSequence:
-                toPrint = False
+            # sam-ls default case: print all items
+            if not args.directories and not args.files and not args.sequences:
+                toPrint = True
+            else:
+                # sam-ls -d
+                if args.directories and itemType == sequenceParser.eTypeFolder:
+                    toPrint = True
+                # sam-ls -f
+                if args.files and itemType == sequenceParser.eTypeFile:
```

Comment:
You are right: it is working because __itemType__ has only one value.
I overwrite the commit: f508ca92a25ca2adfca017515040184702b3eab0